File: VB/03_Slides/Transitions/SetTransitionEffects.vb
Imports System.ComponentModel
Imports System.Text
Imports Spire.Presentation
Imports Spire.Presentation.Drawing.Transition

Namespace SetTransitionEffects
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
			'Create PPT document
			Dim presentation As New Presentation()

			'Load the PPT
			presentation.LoadFromFile("..\..\..\..\..\..\..\Data\SetTransitions.pptx")

			' Set effects
			presentation.Slides(0).SlideShowTransition.Type = TransitionType.Cut
			CType(presentation.Slides(0).SlideShowTransition.Value, OptionalBlackTransition).FromBlack = True

			Dim result As String = "SetTransitionEffects_result.pptx"
			'Save the file
			presentation.SaveToFile(result, FileFormat.Pptx2010)

			Viewer(result)
		End Sub

		Private Sub Viewer(ByVal fileName As String)
			Try
				Process.Start(fileName)
			Catch
			End Try
		End Sub

	End Class
End Namespace

File: CS/03_Slides/SetTransitionEffects.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using Spire.Presentation;
using Spire.Presentation.Drawing.Transition;

namespace SetTransitionEffects
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();            
        }

        private void btnRun_Click(object sender, EventArgs e)
        {
            //Create PPT document
            Presentation presentation = new Presentation();

            //Load the PPT
            presentation.LoadFromFile(@"..\..\..\..\..\..\..\Data\SetTransitions.pptx");

            // Set effects
            presentation.Slides[0].SlideShowTransition.Type = TransitionType.Cut;
            ((OptionalBlackTransition)presentation.Slides[0].SlideShowTransition.Value).FromBlack = true;
          
            String result = "SetTransitionEffects_result.pptx";
            //Save the file
            presentation.SaveToFile(result, FileFormat.Pptx2010);

            Viewer(result);
        }

        private void Viewer(string fileName)
        {
            try
            {
                System.Diagnostics.Process.Start(fileName);
            }
            catch { }
        }

    }
}